cpp/src/groupby/groupby.cu

```diff
@@ -117,6 +117,14 @@ struct empty_column_constructor {
         0, make_empty_column(type_to_id<offset_type>()), empty_like(values), 0, {});
     }
 
+    if constexpr (k == aggregation::Kind::M2 || k == aggregation::Kind::MERGE_M2) {
+      std::vector<std::unique_ptr<column>> child_columns;
```

Comment:
:+1: 
I'm wondering whether this can be phrased differently, as you have suggested before:
```c++
auto begin = cudf::make_counting_transform_iterator(0, [](auto i){ return make_empty_column(FLOAT64); });
return make_structs_column(0, std::vector<std::unique_ptr<column>>(begin, begin+4), 0, {});
```